src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexLWCGCompiler.cs

```diff
@@ -30,6 +31,7 @@ internal sealed class RegexLWCGCompiler : RegexCompiler
         private static int s_regexCount;
 
         /// <summary>The top-level driver. Initializes everything then calls the Generate* methods.</summary>
+        [RequiresDynamicCode("The native code for Regex compilation might not be available at runtime.")]
```

Comment:
I'm late to this. If a method uses, eg. "DynamicMethod" why isn't `[RequiresDynamicCode]` on by default? Is it because we choose to assume that they've guarded the call?

Reply:
Right, I'm asking whether the tool that reads these annotations could look in the body of the method, see it uses DynamicMethod, and figure out for itself that it probably requires dynamic code, without needing annotations.

Reply:
We do have a accompanying code fixer to the analyzer that can do that - give an option to the developer to suppress or add the RDC to the method. Stephen had a suggestion above to not annotate internal call chains when the top level caller has a suppression and I created an issue for that,